src/index.ts

```diff
@@ -484,4 +483,29 @@ export class CruxClient {
         return (this._configService.reverseClientAssetMapping as configurationService.IReverseClientAssetMapping)[assetId];
     }
 
+    private _setKeyPair = async (privateKey?: string): Promise<void> => {
+        if (privateKey) {
+            this._keyPair = JSON.stringify(await this._encryption.encryptJSON(utils.getKeyPairFromPrivKey(privateKey), await this._getEncryptionKey()));
+            this._authenticated = true;
+        } else {
+            this._authenticated = false;
+        }
+        return;
+    }
+
+    private _encryptKeyPair = async () => {
+        // assuming that the decrypted keyPair is stored as an object
+        if (typeof this._keyPair === "object") {
+            this._keyPair = JSON.stringify(await this._encryption.encryptJSON(this._keyPair, await this._getEncryptionKey()));
+        }
+    }
+
+    private _decryptKeyPair = async () => {
+        // assuming that the encrypted keyPair is stored as an object
```

Comment:
comment and code aren't conveying the same behavior.

Reply:
removed them